src/chrdev.rs

```diff
@@ -99,6 +99,29 @@ impl Drop for Registration {
     }
 }
 
+pub struct File {
+    ptr: *const bindings::file,
+}
+
+impl File {
+    fn from_ptr(ptr: *const bindings::file) -> File {
+        return File { ptr };
+    }
+
+    pub fn pos(&self) -> u64 {
+        unsafe {
+            return (*self.ptr).f_pos as u64;
```

Comment:
Doesn't this have lifetime problems? i.e., you can safely call File::from_ptr(ptr), keep it around after ptr goes away, and call pos() later?

Actually you can also safely call `File::from_ptr(1 as *const _).pos()`, no? I can get it to segfault in play.r-l.o :)

If we're enforcing safety by `from_ptr` being private, I'd like to see at least a comment—though also you're calling it in `unsafe` context so you might as well just mark this unsafe. (That is, `pos` being safe is enforced via not constructing and handing out ownership of a `File`, and only handing out a reference to it with the right lifetime, so the obligation is on the caller of `File`'s constructor to get this right, i.e., the constructor should be `unsafe`.)

Reply:
Is there some way I can make it lifetime safe? raw pointers don't have lifetimes like references, so I don't think I can meaningfully bound it?